shell/platform/android/io/flutter/plugin/platform/SingleViewPresentation.java

```diff
@@ -0,0 +1,482 @@
+    private @Nullable WindowManager windowManager;
+    private final Context flutterAppWindowContext;
+
+    PresentationContext(
+        Context base,
+        @NonNull WindowManagerHandler windowManagerHandler,
+        Context flutterAppWindowContext) {
+      super(base);
+      this.windowManagerHandler = windowManagerHandler;
+      this.flutterAppWindowContext = flutterAppWindowContext;
+    }
+
+    @Override
+    public Object getSystemService(String name) {
+      if (WINDOW_SERVICE.equals(name)) {
+        if (isCalledFromAlertDialog()) {
+          // Alert dialogs are showing on top of the entire application and should not be limited to
+          // the virtual
+          // display. If we detect that an android.app.AlertDialog constructor is what's fetching
+          // the window manager
+          // we return the one for the application's window.
+          //
+          // Note that if we don't do this AlertDialog will throw a ClassCastException as down the
+          // line it tries
+          // to case this instance to a WindowManagerImpl which the object returned by
+          // getWindowManager is not
+          // a subclass of.
+          return flutterAppWindowContext.getSystemService(name);
+        }
+        return getWindowManager();
+      }
+      return super.getSystemService(name);
+    }
+
+    private WindowManager getWindowManager() {
+      if (windowManager == null) {
+        windowManager = windowManagerHandler.getWindowManager();
+      }
+      return windowManager;
+    }
+
+    private boolean isCalledFromAlertDialog() {
+      StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
+      for (int i = 0; i < stackTraceElements.length && i < 11; i++) {
+        if (stackTraceElements[i].getClassName().equals(AlertDialog.class.getCanonicalName())
+            && stackTraceElements[i].getMethodName().equals("<init>")) {
+          return true;
+        }
+      }
+      return false;
+    }
+  }
+
+  /*
+   * A dynamic proxy handler for a WindowManager with custom overrides.
+   *
+   * The presentation's window manager delegates all calls to the default window manager.
+   * WindowManager#addView calls triggered by views that are attached to the virtual display are crashing
+   * (see: https://github.com/flutter/flutter/issues/20714). This was triggered when selecting text in an embedded
+   * WebView (as the selection handles are implemented as popup windows).
+   *
+   * This dynamic proxy overrides the addView, removeView, removeViewImmediate, and updateViewLayout methods
+   * to prevent these crashes.
+   *
+   * This will be more efficient as a static proxy that's not using reflection, but as the engine is currently
+   * not being built against the latest Android SDK we cannot override all relevant method.
+   * Tracking issue for upgrading the engine's Android sdk: https://github.com/flutter/flutter/issues/20717
```

Comment:
This is an old issue and the Andorid SDK has long since been updated a few times over. If we can get rid of some of the reflection here we should.

Reply:
similar comment. This could be a separate PR

Reply:
https://github.com/flutter/flutter/issues/106449